doc on upgrading to v1.0

skriss opened the pull request:
Signed-off-by: Steve Kriss <krisss@vmware.com>

@nrb @carlisia 

This will replace the existing "Upgrade from Ark to Velero" doc - I need to decide exactly how to rejigger the TOC on the docs site - may get rid of the Upgrade section and just move this into Install or About.

carlisia commented:
Isn't this structure all going to change with the imminent website?

carlisia commented:
I have suggestions for a different break down of the content for your consideration.

- I would put the pre-requisite section at the very top.
- I would delete the first paragraph.
- I could cut the 2nd paragraph and bullets (maybe it can be reused below).

Then I would split the instruction section into:
Upgrading from v0.11.1
- Description of what will be done.
- Instructions

Upgrading to v0.11.1 (Alternatively, could be titled "Upgrading from versions earlier than v0.11.1" or something like that.
"Maybe" add something brief about the need to upgrade to v0.11.1 before upgrading to v1.0. Maybe the content below will be self-evident.
- Description of what will be done.
- Instructions

I *think* this works. The intent is to guide people more obviously to the use case they find themselves in.

skriss commented:
I pushed an update which I *think* mostly addresses your comments. I left the order of the two parts as-is since they need to be executed in that order.  Let me know if I didn't address anything specific that you still think needs to change!

carlisia commented:
If you are already on version 0.11.1, do you need to do part 1?

skriss commented:
It's not about what version of Velero you're currently running, it's about whether you have backups around that were originally created prior to v0.11.

carlisia commented:
Ah yeah, I hadn't totally grasped this fact. Love it that you added `While Velero v0.11 is backwards-compatible with these files, Velero v1.0 is not.`.

This structure looks good. Gonna add 1 tiny comment.

carlisia commented on docs/upgrade-to-1.0.md:
```diff
@@ -0,0 +1,85 @@
+
+#### Overview
+
+If you have any backups that were originally taken with a version prior to v0.11 (i.e. when the project was named Ark), you need to replace the legacy backup metadata in object storage with updated versions. While Velero v0.11 is backwards-compatible with these files, Velero v1.0 is not.
```
`s/If you have any backups/In case` you have any backups.

For some reason this sounds more like a prompt to be cautious and run this part 1.

skriss replied:
that edit sounds slightly strange to my ear, but i pushed an update that i think accomplishes the same thing.

carlisia replied:
Yes, looks great now.

carlisia commented:
Yes, looks great now.

skriss commented:
squashed, still need to decide on TOC update.

carlisia commented:
I'd move it to the install page and on that page have a TOC 

- Install
- Update

then the sections.

skriss commented:
@nrb this is ready to merge if you're happy with it. I'll do updates on `gh-pages` next.
